addon/services/key-manager.js

```diff
@@ -157,43 +156,28 @@ export default Ember.Service.extend({
   _combosWithKeys(combos) {
     const downKeys = get(this, 'downKeys');
     return combos.filter((combo) => {
-      const keys = get(combo, 'keys');
+      const keys = get(combo, 'keys').slice();
+
+      const verifyModifiers = modifierKeys.every((key) => {
+        return !keys.includes(key) || get(this, `${key}Key`);
+      });
+      keys.removeObjects(modifierKeys);
+
       const sameLength = keys.length === downKeys.length;
-      return sameLength && keys.every((key) => {
+      const isMatch = keys.every((key) => {
         return downKeys.includes(keyCodes[key]);
       });
-    });
-  },
 
-  _handleVisibilityChange() {
-    if (document.visibilityState === 'hidden') {
-      this._resetDownKeys();
-    }
+      return sameLength &&
+        isMatch &&
+        verifyModifiers;
+    });
   },
 
   _resetDownKeys() {
-    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) {
```

Comment:
I don't think we need the guard since we are not running this on an interval anymore.